modules/core/src/test/java/org/apache/ignite/internal/processors/cache/CacheDataRegionConfigurationTest.java

```diff
@@ -169,4 +173,157 @@ public void testProperlySizedMemoryPolicy() throws Exception {
             fail("With properly sized DataRegion no exceptions are expected to be thrown.");
         }
     }
+
+    /**
+     * Verifies that {@link IgniteCheckedException} is thrown when swap and persistence are enabled at the same time
+     * for a data region.
+     */
+    public void testSetPersistenceAndSwap() {
+        DataRegionConfiguration invCfg = new DataRegionConfiguration();
+
+        invCfg.setName("invCfg");
+        invCfg.setInitialSize(DFLT_MEM_PLC_SIZE);
+        invCfg.setMaxSize(DFLT_MEM_PLC_SIZE);
+        // Enabling the persistence.
+        invCfg.setPersistenceEnabled(true);
+        // Enabling the swap space.
+        invCfg.setSwapPath("/path/to/some/directory");
+
+        memCfg = new DataStorageConfiguration();
+        memCfg.setDataRegionConfigurations(invCfg);
+
+        ccfg = new CacheConfiguration(DEFAULT_CACHE_NAME);
+        ccfg.setDataRegionName("ccfg");
+
+        GridTestUtils.assertThrows(log(), new Callable<Object>() {
+            @Nullable @Override public Object call() throws Exception {
+                startGrid(0);
+                return null;
+            }
+        }, IgniteCheckedException.class, "Failed to start processor: GridProcessorAdapter []");
+    }
+
+    /**
+     * Verifies that {@link IgniteCheckedException} is thrown when page eviction threshold isn't between 0.5 and 0.999.
+     */
+    public void testSetInvalidEviction() {
+        final double SMALL_EVICTION_THRESHOLD = 0.1D;
+        final double BIG_EVICTION_THRESHOLD = 1.0D;
+        DataRegionConfiguration invCfg = new DataRegionConfiguration();
+
+        invCfg.setName("invCfg");
+        invCfg.setInitialSize(DFLT_MEM_PLC_SIZE);
+        invCfg.setMaxSize(DFLT_MEM_PLC_SIZE);
+        invCfg.setPageEvictionMode(DataPageEvictionMode.RANDOM_LRU);
+        // Setting the page eviction threshold less than 0.5
+        invCfg.setEvictionThreshold(SMALL_EVICTION_THRESHOLD);
+
+        memCfg = new DataStorageConfiguration();
+        memCfg.setDataRegionConfigurations(invCfg);
+
+        ccfg = new CacheConfiguration(DEFAULT_CACHE_NAME);
+
+        GridTestUtils.assertThrows(log(), new Callable<Object>() {
+            @Nullable @Override public Object call() throws Exception {
+                startGrid(0);
+                return null;
+            }
+        }, IgniteCheckedException.class, "Failed to start processor: GridProcessorAdapter []");
+
+        // Setting the page eviction threshold greater than 0.999
+        invCfg.setEvictionThreshold(BIG_EVICTION_THRESHOLD);
+        memCfg.setDataRegionConfigurations(invCfg);
+
+        GridTestUtils.assertThrows(log(), new Callable<Object>() {
```

Comment:
I'd rather have one `assertThrows` per test, to have the checks run independently. In it cumbersome in JUnit 3, but I'd usually do it like
```
public void testA() { check("a"); }
public void testB() { check("b"); }
private void check(String param) { /* actual test code goes here */ }
```

Reply:
Ok I will fix.

Reply:
Updated.